src/metabase/task/persist_refresh.clj

```diff
@@ -236,14 +240,16 @@
 
 (defn- cron-schedule
   "Return a cron schedule that fires every `hours` hours."
-  [hours]
-  (cron/schedule
-   ;; every 8 hours
-   (cron/cron-schedule (if (= hours 24)
-                         ;; hack: scheduling for midnight UTC but this does raise the issue of anchor time
-                         "0 0 0 * * ? *"
-                         (format "0 0 0/%d * * ? *" hours)))
-   (cron/with-misfire-handling-instruction-do-nothing)))
+  [hours anchor-time]
+  (let [[start-hour start-minute] (map parse-long (str/split anchor-time #":"))]
+    (cron/schedule
+      (cron/cron-schedule (if (= 24 hours)
+                            (format "0 %d %d * * ? *" start-minute start-hour)
+                            (format "0 %d %d/%d * * ? *" start-minute start-hour hours)))
```

Comment:
One thing i'm not sure how it works, if you want an anchor time of 9am and then do an interval of 23 hours, what time will this run on day 2? The anchor time and interval aren't in harmony.
- if the interval matters more, 9 am and 23 hour schedule means day 2 the sync will run at 8 am, then 7 am, etc.
- if the anchor time matters more, it will be equivalent to once a day at 9 am.

Do we know which scenario will happen in this case?

Reply:
This would run once a day at 9am, the anchor time determines the first run time of the day, then after X hours **for that day** it will run again. At midnight, nothing runs until the anchor time.